multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/BuildCloudUndeployModelStep.java

```diff
@@ -98,10 +98,11 @@ protected String getStepErrorMessage(ProcessContext context) {
                                    .collect(Collectors.toList());
     }
 
-    private List<String> getServicesToCreate(ProcessContext context) {
-        return context.getVariable(Variables.SERVICES_TO_CREATE)
+    private List<String> getServicesFromBatches(ProcessContext context) {
```

Comment:
A more appropriate name for this method would be `getAllServiceNames` or something of the sort. You are getting all services from all batches.